learningequality/studio
# More completion duration updates

## Summary
### Description of the change(s) you made
Cohack with @rtibbles that began as changes to address [feedback](https://github.com/learningequality/studio/pull/3777#issuecomment-1295067138), but resulted in some broader refactoring.

- Completion and duration should now work for _all_ content types 
- The descriptive aspect of duration has been hidden for now, so duration setting is only showed if a time based completion criterion is set. This was done due to the confusing interface language in the case that the it was being set descriptively.
- The learner managed state is moved into the completion options component, so that when it is set when no other completion criterion is set, the default completion criteria state to ensure validity.
- To simplify frontend state management, backend completion criteria validation is only run when the node is set as complete, allowing saving of partial completion criteria

Issue for Kolibri filed separately in Kolibri repo 

### Manual verification steps performed
1. many many many permutations of testing
2. test each content type
3. test all completion durations options of content type
4. switch from each completion option to each other option (i.e from reference to time-based, reference to all content viewed, time based to all content viewed, time based to reference, all content viewed to reference, all content viewed to time-based) 
5. ensure "learner can mark as complete" works without any other interactions with completion/duration (i.e. check the checkbox, but make no other changes to default state)


### Screenshots (if applicable)
toooo many to add 

### Does this introduce any tech-debt items?

___
## Reviewer guidance
### How can a reviewer test these changes?



### Are there any risky areas that deserve extra testing?



## References


## Comments


----

### Contributor's Checklist


PR process:


Studio-specifc:



Testing:

___

### Reviewer's Checklist
#### This section is for reviewers to fill out.
